### PR TITLE
Reports: Add grouping option in Proof of Play

### PR DESCRIPTION
**Pull request by mgbaybay**
## Changes
FE:
- Added `Group By` filter options
- Show and hide columns based on group by selection

BE:
- [mySQL and MongoDB] Added data calculation for accumulating the values according to grouping option

Relates to: https://github.com/xibosignage/xibo/issues/3356

### CODE REVIEW

File: reports/proofofplay-report-form.twig [hunk 1]
```diff
@@ -80,6 +80,14 @@
                             {% set title %}{% trans "Time" %}{% endset %}
                             {{ inline.time("statsToDtTime", title, "00:00", "", "stats-to-dt-time") }}
 
+                            {% set title %}{% trans "Group By" %}{% endset %}
+                            {% set options = [
+                                { id: "", name: "" },
```
Comment (dasgarner) [inline reports/proofofplay-report-form.twig:85]: Can we call this option "Display" ?

**Review by dasgarner — COMMENTED** [review]
I have a few early pieces of feedback - although maybe I misunderstood

File: reports/proofofplay-report-form.twig (hunk 1 above)
Comment (mgbaybay) [reply reports/proofofplay-report-form.twig]: Yup! Added `display` as a default option. Thank you for the review!

File: lib/Report/ProofOfPlay.php [hunk 2]
```diff
@@ -555,12 +576,16 @@ private function getProofOfPlayReportMySql(
...
+        $logicalOperator,
+        $groupBy,
+        $displayGroupIds
     ) {
         $fromDt = $fromDt->format('U');
         $toDt = $toDt->format('U');
 
         // Media on Layouts Ran
+
+//        $select = 'SELECT `displaygroup`.`displaygroup`, `displaygroup`.`displayGroupId` from displaygroup';
```
Comment (dasgarner) [inline lib/Report/ProofOfPlay.php:588]: Did you mean to leave this in here?

File: lib/Report/ProofOfPlay.php [hunk 3]
```diff
@@ -801,21 +838,25 @@ private function getProofOfPlayReportMySql(
 
         $body .= '
             GROUP BY stat.type, 
-                stat.tag, 
-                display.Display, 
                 stat.parentCampaignId,
                 stat.displayId,
```
Comment (dasgarner) [inline lib/Report/ProofOfPlay.php:842]: We are still grouping by `stat.displayId`

**Review by dasgarner — CHANGES_REQUESTED** [review]

**Comment by mgbaybay** [comment]
Closing this to avoid conflicts with the updated source branch

**Pull request closed without merging**